packages/core/src/button/button.element.ts

```diff
@@ -80,28 +80,36 @@ export class CdsButton extends CdsBaseButton {
    * - `success`: disables the button and shows a check mark inside the button; auto-triggers to change back to DEFAULT state after 1000 ms
    * - `error`: shows the content of the button (in the context of application, this state is usually entered from a LOADING state. the application should show appropriate error message)
    */
-  @property({ type: String }) get loadingState() {
-    return this._loadingState;
-  }
+  @property({ type: String })
+  loadingState: keyof typeof ClrLoadingState = ClrLoadingState.default;
+
+  firstUpdated(props: Map<string, any>) {
+    super.firstUpdated(props);
 
-  set loadingState(value: keyof typeof ClrLoadingState) {
-    // track prior disabled state to set prior value after button is re-enabled from a loading state
-    if (this._loadingState === ClrLoadingState.default) {
-      this._disabled = this.disabled;
+    if (this.loadingState !== ClrLoadingState.default) {
+      this.disabled = true;
     }
+  }
+
+  update(props: Map<string, any>) {
+    if (props.has('loadingState')) {
+      if (props.get('loadingState') === ClrLoadingState.default) {
+        // track prior disabled state to set prior value after button is re-enabled from a loading state
+        this._disabled = this.disabled;
+      }
 
-    if (value === ClrLoadingState.default) {
-      this.enableButton();
-    } else {
-      this.disableButton();
+      if (props.get('loadingState') !== undefined) {
+        if (this.loadingState === ClrLoadingState.default) {
```

Comment:
Can we move these duplicated checks into a getter? Something like `isLoading`?

Reply:
I ask so that we can keep the code DRY. In the event that a further step is needed in determining if the button is loading, we would need to make changes in three places instead of one...

Reply:
Updated, the other states aren't loading, even though the button is disabled for them, so I went with the very literal `isDefaultLoadingState`